spring-cloud-sleuth-api/src/main/java/org/springframework/cloud/sleuth/Tag.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth;
+
+/**
+ * Key/value pair representing a dimension of a span used to classify and drill into
+ * measurements. Inspired by Micrometer.
+ *
+ * @author Marcin Grzejszczak
+ * @since 3.0.3
+ */
+public interface Tag extends Comparable<Tag> {
+
+	/**
+	 * @return tag key
+	 */
+	TagKey getTagKey();
+
+	/**
+	 * @return tag value
+	 */
+	String getValue();
+
+	/**
+	 * @return additional tag description
+	 */
+	default String getDescription() {
+		return "";
+	}
+
+	/**
+	 * Will apply the tag for the given key and value.
+	 * @param span to tag
+	 * @return tagged span
+	 */
+	default Span tag(Span span) {
```

Comment:
I'm not sure about the `applyOn(...)` since what we're doing is we're tagging a span, builder or whatever else. `tag` is shorter than `applyOn`. WDYT?

Reply:
With the wrapping logic, this comment is now invalid.